### ISSUE TITLE
can't get anaconda to work

### ISSUE THREAD

**Issue by mladimir** [DamnWidget/anaconda#707]
I am terribly new to all of this and I am having some issues.

## Expected Behaviour
All the Anaconda functionality.
## Actual Behaviour
Nothing happens. No linting, no formatting even when I right click> Anaconda and auto format nothing happens
## Steps to Reproduce
I just installed Anaconda with package control. I've installed python with the Anaconda3 distribution and also enabled the `PATH`  environment variables. I can compile python scripts in ST3 just fine.
## ST3, Anaconda and OS versions
version 3.143 // 2.1.22 // windows 10
## ST3 Console Logs
The python interpreter path is set at

```
>>> get_settings(view, 'python_interpreter')
'python'
```
I get these kind of errors in the console:

```
Traceback (most recent call last):
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 389, in run_callback
    expr()
    run_callback('on_activated', callback, lambda: callback.on_activated(v))
  File "C:\Users\Milad\AppData\Roaming\Sublime Text 3\Packages\Anaconda\listeners\linting.py", line 124, in on_activated
    self.run_linter(view)
  File "C:\Users\Milad\AppData\Roaming\Sublime Text 3\Packages\Anaconda\anaconda_lib\linting\sublime.py", line 380, in run_linter
    Worker().execute(Callback(on_success=parse_results), **data)
  File "C:\Users\Milad\AppData\Roaming\Sublime Text 3\Packages\Anaconda\anaconda_lib\workers\market.py", line 86, in execute
    worker = cls.hire(cls)
  File "C:\Users\Milad\AppData\Roaming\Sublime Text 3\Packages\Anaconda\anaconda_lib\workers\market.py", line 32, in hire
    itprt = Interpreter(raw_interpreter)
  File "C:\Users\Milad\AppData\Roaming\Sublime Text 3\Packages\Anaconda\anaconda_lib\workers\interpreter.py", line 26, in __init__
    self.__parse_raw_interpreter()
  File "C:\Users\Milad\AppData\Roaming\Sublime Text 3\Packages\Anaconda\anaconda_lib\workers\interpreter.py", line 192, in __parse_raw_interpreter
    return self.__prepare_local_interpreter()
  File "C:\Users\Milad\AppData\Roaming\Sublime Text 3\Packages\Anaconda\anaconda_lib\workers\interpreter.py", line 85, in __prepare_local_interpreter
    self.__extract_paths(view)
  File "C:\Users\Milad\AppData\Roaming\Sublime Text 3\Packages\Anaconda\anaconda_lib\workers\interpreter.py", line 129, in __extract_paths
    extra = get_settings(view, 'extra_paths', [])
  File "C:\Users\Milad\AppData\Roaming\Sublime Text 3\Packages\Anaconda\anaconda_lib\helpers.py", line 257, in get_settings
    r = os.path.expanduser(os.path.expandvars(r))
  File "./python3.3/ntpath.py", line 311, in expanduser
AttributeError: 'list' object has no attribute 'startswith'
```
## Anaconda's JsonServer Logs
there is no logs folder in `\AppData\Roaming\Sublime Text 3\Packages\Anaconda`

**Comment by sametmax** [comment]
I have exactly the same behaviour, on the latest Sublime version, but on Ubuntu.

The error is slightly different since it's using posixpath instead of ntpath but in it's basically identical:

```
  File "/home/user/Scripts/stow/home_dotfiles/.config/sublime-text-3/Packages/Anaconda/anaconda_lib/workers/interpreter.py", line 129, in __extract_paths
    extra = get_settings(view, 'extra_paths', [])
  File "/home/user/Scripts/stow/home_dotfiles/.config/sublime-text-3/Packages/Anaconda/anaconda_lib/helpers.py", line 257, in get_settings
    r = os.path.expanduser(os.path.expandvars(r))
  File "./python3.3/posixpath.py", line 252, in expanduser
AttributeError: 'list' object has no attribute 'startswith'
```

The problem comes from the fact that the variable r contains a list here:

https://github.com/DamnWidget/anaconda/blob/master/anaconda_lib/helpers.py#L256

The last update added that line:

https://github.com/DamnWidget/anaconda/commit/1d52726fb55f03da9626048822691794ef21c4b2#diff-19aa897b833995404fa88d3dbc5fa8a7

I tried to replace it with:

`r = [os.path.expanduser(os.path.expandvars(v)) for v in r]`

But then got:

 ```
 File "/home/user/Scripts/stow/home_dotfiles/.config/sublime-text-3/Packages/Anaconda/anaconda_lib/workers/interpreter.py", line 185, in __parse_raw_interpreter
    urldata = urlparse(self.__raw_interpreter)
  File "./python3.3/urllib/parse.py", line 292, in urlparse
  File "./python3.3/urllib/parse.py", line 112, in _coerce_args
  File "./python3.3/urllib/parse.py", line 96, in _decode_args
  File "./python3.3/urllib/parse.py", line 96, in <genexpr>
AttributeError: 'list' object has no attribute 'decode'

So the error is probably deeper than just a line added.

Now what's worries me is that this error probably broke ALL anaconda setup with automatic updates. Which means the project has no basic unit test to prevent releasing something that is not completely broken. It's usually a bad thing, but with auto-update, it's especially bad.

As much as I like anaconda, it's not the first time I have to spent half an hour to figure out why it's not working anymore. Can I make a donation somewhere to help you so you can spend more time on code quality ?
```

**Comment by DamnWidget** [comment]
Woopsie, looks like I totally broke it, give me some mins guys.

**Comment by mladimir** [comment]
@sametmax 
Thanks a lot for the prompt reply! Yes that did the job for now.
Guess I just had bad luck starting to code python in ST today, hehe.

@DamnWidget
Thank you for taking the time to work on this package freely. Much appreciated.

**Comment by DamnWidget** [comment]
Fixed on [v2.1.23|https://github.com/DamnWidget/anaconda/releases/tag/v2.1.23]

By the way, @sametmax, the project has unit tests, it also has a travis hook that runs on each push but that unit tests are just for the JsonServer part, add unit testing for anything that is related with the ST3 embedded interpreter is horribly complex and time consuming there had been many attempts from part of the ST3 plugins development community to create a package to make the task easier but it is still a pain  in the bottom so we just don't have unit tests for client side code (client side is any code that is executed by the ST3 embedded python interpreter).

Said that, PRs are welcome.

**Comment by DamnWidget** [comment]
Closing this @mladimir feel free to open another one if you find any issues.
